### PR TITLE
Makes foam less garbage

### PR DESCRIPTION
**Pull request by Iamgoofball**
only took me 6 months and phil saying I can't make any new PRs until it's fixed

foam now only divides by 7, and it won't apply the reagent's effects unless the lifetime is a multiple of 7

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Bring back foamestry goof

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -53,22 +54,24 @@
 		kill_foam()
 		return
 
-	var/fraction = 1/initial(lifetime)
+	var/fraction = 1/initial(reagent_divisor)
```
Comment (optimumtact) [inline code/game/objects/effects/effect_system/effects_foam.dm]: you should sanity check the fraction is an integer and probably also that it's >= 0

Comment (Iamgoofball) [inline code/game/objects/effects/effect_system/effects_foam.dm]: is there a proc for this or do I need to do it manually?

**Comment by phil235** [comment]
I hope you tested it, @optimumtact

**Pull request merged**